aptos-move/parallel-executor/src/executor.rs

```diff
@@ -323,25 +311,22 @@ where
             }
         });
 
-        // Extract outputs in parallel.
-        let num_txns = scheduler.num_txn_to_execute();
-        let valid_results_size = AtomicUsize::new(num_txns);
-        let chunk_size = (num_txns + 4 * self.concurrency_level - 1) / (4 * self.concurrency_level);
-        RAYON_EXEC_POOL.install(|| {
-            (0..num_txns)
-                .collect::<Vec<TxnIndex>>()
-                .par_chunks(chunk_size)
-                .map(|chunk| {
-                    for idx in chunk.iter() {
-                        let res = last_input_output.take_output(*idx);
-                        if matches!(res, ExecutionStatus::SkipRest(_)) {
-                            valid_results_size.fetch_min(*idx + 1, Ordering::SeqCst);
-                        }
-                        outcomes.set_result(*idx, res);
-                    }
-                })
-                .collect::<()>();
-        });
+        // TODO: for large block sizes and many cores, extract outputs in parallel.
+        let mut maybe_err = None;
+        let final_results = (0..scheduler.num_txn_to_execute())
+            .collect::<Vec<TxnIndex>>()
+            .iter()
```

Comment:
using for loop seems easier than the iter, we can short circuit to return error/skip rest

Reply:
Done